### PR TITLE
#116 use reducer in auth context

### LINKED ISSUE #116: Client: AuthContext should use a reducer instead of states
Instead of using different states for the authContext a single object with all necessary data should be used.
With a reducer instead of a state, the logic of changing the auth object is kept in the authContext itself which makes the use of it easier.

### PR DESCRIPTION
**Pull request by LennartRukower**
Refactored AuthContext to use reducer. Also added global types file to contain action and reducer types. 

Closes #116

### CODE REVIEW

File: client/src/ts/members/membersTypes.ts [hunk 1]
```diff
@@ -6,265 +6,263 @@
...
  */
 export type EDVSkill = {
-    /*
-     * Additional attribute which is needed for the Automplece Component
-     * It is only used for filtering
-     */
-    inputValue?: string;
-    wert: string;
+  /*
+   * Additional attribute which is needed for the Automplece Component
```
Comment (christophtrybek) [inline client/src/ts/members/membersTypes.ts:38]: I guess it is the "Autocomplete Component" and not an "Automplece Component"

File: client/src/ts/members/membersTypes.ts [hunk 2]
```diff
@@ -6,265 +6,263 @@
...
  */
 export type Language = {
-    /*
-     * Additional attribute which is needed for the Automplece Component
-     * It is only used for filtering
-     */
-    inputValue?: string;
-    wert: string;
+  /*
+   * Additional attribute which is needed for the Automplece Component
```
Comment (christophtrybek) [inline client/src/ts/members/membersTypes.ts:26]: I guess it is the "Autocomplete Component" and not an "Automplece Component"

**Review by christophtrybek — APPROVED** [review]
LGTM

**Review by jbt-it — APPROVED** [review]
LGTM

**Pull request merged**